functions/helloworld/hello_http_system_test.go

```diff
@@ -20,20 +20,82 @@
 package helloworld
 
 import (
+	"fmt"
 	"io/ioutil"
+	"log"
 	"net/http"
 	"net/url"
 	"os"
+	"os/exec"
 	"strings"
 	"testing"
 	"time"
 )
 
+const RuntimeVersion = "go113"
+
+func TestMain(m *testing.M) {
+	// Only run end-to-end tests when configured to do so.
+	if os.Getenv("GOLANG_SAMPLES_E2E_TEST") == "" {
+		log.Println("Skipping end-to-end tests: GOLANG_SAMPLES_E2E_TEST not set")
+		os.Exit(m.Run())
+	}
+
+	retn, err := setupAndRun(m)
+	if err != nil {
+		log.Fatal(err)
+	}
+	os.Exit(retn)
+}
+
+func setupAndRun(m *testing.M) (int, error) {
+	entryPoint := "HelloHTTP"
+	name := entryPoint + "-" + time.Now().Format("20060102-150405")
+
+	// Setup function for tests.
+	cmd := exec.Command("gcloud", "functions", "deploy", name,
+		"--entry-point="+entryPoint,
+		"--runtime="+RuntimeVersion,
+		"--allow-unauthenticated",
+		"--trigger-http",
+	)
+	log.Printf("Running: %s %s", cmd.Path, strings.Join(cmd.Args, " "))
+	if _, err := cmd.Output(); err != nil {
+		log.Println(string(err.(*exec.ExitError).Stderr))
+		return 1, fmt.Errorf("Setup: Deploy function: %w", err)
+	}
+
+	// Tear down the deployed function.
+	defer func() {
+		cmd = exec.Command("gcloud", "functions", "delete", name)
+		log.Printf("Running: %s %s", cmd.Path, strings.Join(cmd.Args, " "))
+		if _, err := cmd.Output(); err != nil {
+			log.Println(string(err.(*exec.ExitError).Stderr))
+			log.Printf("Teardown: Delete function: %v", err)
+		}
+	}()
+
+	// Retrieve URL for tests.
+	cmd = exec.Command("gcloud", "functions", "describe", name, "--format=value(httpsTrigger.url)")
+	log.Printf("Running: %s %s", cmd.Path, strings.Join(cmd.Args, " "))
+	out, err := cmd.Output()
+	if err != nil {
+		log.Println(string(err.(*exec.ExitError).Stderr))
+		return 1, fmt.Errorf("Setup: Get function URL: %w", err)
+	}
+	if err := os.Setenv("BASE_URL", strings.TrimSpace(string(out))); err != nil {
+		return 1, fmt.Errorf("Setup: os.Setenv: %w", err)
+	}
+
+	// Run the tests.
+	return m.Run(), nil
+}
+
 func TestHelloHTTPSystem(t *testing.T) {
 	client := http.Client{
 		Timeout: 10 * time.Second,
 	}
-	urlString := os.Getenv("BASE_URL") + "/HelloHTTP"
+	urlString := os.Getenv("BASE_URL")
```

Comment:
I think `BASE_URL` intentionally didn't include the name of the function. Maybe it shouldn't? But, we'd need to update the docs here: https://cloud.google.com/functions/docs/testing/test-http#functions-testing-http-system-cmd-go

Reply:
I'm a bit confused what our goal is here. I'm not putting this together as a sample of a system test, but an internal system test. I'm arranging it so the full URL comes from a gcloud command, rather than based on the user's understanding of the URL pattern.

Reply:
+1 to @tbpg here. If we plan on updating any _public-facing_ samples in a similar manner, we'll have to update this for the [other languages](https://cloud.google.com/functions/docs/testing/test-http#system_tests) as well.

(@grayside if these samples are _not_ public-facing, perhaps we should use a different env. var name here - to avoid confusion?)

Reply:
Resolved in discussion, going to back out this piece in the expectation that URL pattern changes will then break the tests.